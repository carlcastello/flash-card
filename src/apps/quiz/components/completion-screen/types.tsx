export interface IOwnProps {
  onClose: () => void,
  onReview: () => void,
  classes: {
    
  }
}