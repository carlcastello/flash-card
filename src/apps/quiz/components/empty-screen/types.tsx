export interface IOwnProps {
  onEdit: () => void,
  onClose: () => void,
  classes: {

  }
}