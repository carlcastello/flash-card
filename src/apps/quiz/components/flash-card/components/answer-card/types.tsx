import { FlashcardStatus } from "../../types";

export interface IOwnProps {
  answer: string,
  flashcardStatus: FlashcardStatus,
  ref: any,
  classes: {
    paper: string,
    input: string,
    typography: string
  }
}

export interface IOwnState {
  answer: string,
  hideAnswer: boolean,
}