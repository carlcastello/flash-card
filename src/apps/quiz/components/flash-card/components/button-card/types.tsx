import { FlashcardStatus } from "../../types";

export interface IOwnProps {
  flashcardStatus: FlashcardStatus,
  submit: () => void,
  next: () => void,
  classes: {
    button: string
  }
}

export interface IOwnState {

}