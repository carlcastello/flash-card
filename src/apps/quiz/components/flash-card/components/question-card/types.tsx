import { FlashCardStatus } from "../../types";
import { QuestionType } from "../../../../../commons/types";

export interface IQuestionObject {
  question: string,
  subQuestion?: string,
  hint?: string,
  questionType: QuestionType,
}

export interface IOwnProps {
  flashCardStatus: FlashCardStatus,
  questionObject: IQuestionObject,
  onSkip: () => void,
  classes: {
    paper: string,
    boxButtonContainer: string,
    boxQuestionaireContainer: string
  }
}