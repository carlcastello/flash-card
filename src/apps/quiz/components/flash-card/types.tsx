import { IFlashCard } from "../../../commons/types";

export enum FlashCardStatus {
  DEFAULT = "DEFAULT",
  CORRECT = "CORRECT",
  WRONG = "WRONG",
  HINT = "HINT"
}

export interface IOwnProps {
  update: (flashCardStatus: FlashCardStatus) => void,
  next: () => void,
  flashCardObject: IFlashCard,
  classes: {

  }
}

export interface IOwnState {
  flashCardStatus: FlashCardStatus
}