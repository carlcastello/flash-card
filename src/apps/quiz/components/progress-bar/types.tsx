import { FlashcardStatus } from "../flash-card/types";

export interface IOwnProps {
  currentQuestion: number,
  totalQuestion: number,
  flashcardStatus: FlashcardStatus,
  classes: {
    title: string,
    progressBarContainer: string,
    progressBar: string,
  }
}