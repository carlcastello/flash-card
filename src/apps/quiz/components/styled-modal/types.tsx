export interface IOwnProps {
  isOpen: boolean,
  onConfirm?: () => void,
  onIgnore: () => void,

  classes: {
    modalContainer: string,
    modalPaperContainer: string,
    closeButton: string,
  }
}