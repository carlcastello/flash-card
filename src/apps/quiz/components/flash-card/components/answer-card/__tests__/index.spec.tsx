import React from "react";
import { shallow } from "enzyme";

import { AnswerCard } from "../index";
import { FlashcardStatus } from "../../../types";
import toJson from "enzyme-to-json";
import { Input, Collapse } from "@material-ui/core";

describe("The AnswerCard", () => {

  let focusFunction: any;
  let blurFunction: any;

  let answer: string = "This is the correct answer";

  let wrapper: any;
  let instance: any;

  beforeEach(() => {
    focusFunction = jest.fn();
    blurFunction = jest.fn();

    wrapper = shallow(
      <AnswerCard
        answer={answer}
        flashcardStatus={FlashcardStatus.DEFAULT}
        ref={jest.fn()}
        classes={{
          paper: "paper",
          input: "string",
          typography: "typography"
        }}/>
    );

    instance = wrapper.instance();
    instance.inputRef = { focus: focusFunction, blur: blurFunction };
  });

  describe("the render funtion", () => {
    it("shows the answer card as FlashcardStatus.DEFAULT", () => {
      expect(toJson(wrapper)).toMatchSnapshot();
      expect(wrapper.find(Collapse).props().in).toEqual(false);
      expect(wrapper.find(Input)).toHaveLength(2);
      expect(wrapper.find(Input).at(0).props().disabled).toEqual(false);
    });

    it("shows the answer casrd as FlashcardStatus.CORRECT", () => {
      wrapper.setProps({ flashcardStatus: FlashcardStatus.CORRECT });

      expect(toJson(wrapper)).toMatchSnapshot();
      expect(wrapper.find(Collapse).props().in).toEqual(false);
      expect(wrapper.find(Input)).toHaveLength(2);
      expect(wrapper.find(Input).at(0).props().disabled).toEqual(true);
    });

    it("shows the correct answer when FlashcardStatus.WRONG", () => {
      wrapper.setState({ hideAnswer: false });
      wrapper.setProps({ flashcardStatus: FlashcardStatus.WRONG });

      expect(toJson(wrapper)).toMatchSnapshot();
      expect(wrapper.find(Collapse).props().in).toEqual(true);
      expect(wrapper.find(Input)).toHaveLength(2);
      expect(wrapper.find(Input).at(0).props().disabled).toEqual(true);
    });

    it("shows the correct answer and removed the input when FlashcardStatus.HINT", () => {
      wrapper.setState({ hideAnswer: false });
      wrapper.setProps({ flashcardStatus: FlashcardStatus.HINT });

      expect(toJson(wrapper)).toMatchSnapshot();
      expect(wrapper.find(Collapse).props().in).toEqual(true);
      expect(wrapper.find(Input)).toHaveLength(1);
    });
  });

  describe("The componentDidUpdate", () => {
    it("calls neither function if currentProps == prevProps", () => {
      expect(focusFunction).toHaveBeenCalledTimes(0);
      expect(blurFunction).toHaveBeenCalledTimes(0);

      wrapper.setProps({ flashcardStatus: FlashcardStatus.DEFAULT });

      expect(focusFunction).toHaveBeenCalledTimes(0);
      expect(blurFunction).toHaveBeenCalledTimes(0);
    });

    it("calls focusFunction if FlashcardStatus.DEFAULT and currentProps != prevProps", () => {
      expect(focusFunction).toHaveBeenCalledTimes(0);
      expect(blurFunction).toHaveBeenCalledTimes(0);

      // Updating the props to rogger on Focus between CORRECT -> DEFAULT
      wrapper.setProps({ flashcardStatus: FlashcardStatus.CORRECT });
      expect(focusFunction).toHaveBeenCalledTimes(0);
      expect(blurFunction).toHaveBeenCalledTimes(1);

      wrapper.setProps({ flashcardStatus: FlashcardStatus.DEFAULT });

      expect(focusFunction).toHaveBeenCalledTimes(1);
      expect(blurFunction).toHaveBeenCalledTimes(1);
    });

    it("calls blurFunction if not FlashcardStatus.DEFAULT and currentProps != prevProps", () => {
      expect(focusFunction).toHaveBeenCalledTimes(0);
      expect(blurFunction).toHaveBeenCalledTimes(0);

      wrapper.setProps({ flashcardStatus: FlashcardStatus.CORRECT });

      expect(focusFunction).toHaveBeenCalledTimes(0);
      expect(blurFunction).toHaveBeenCalledTimes(1);
    });
  });

  describe("The state function", () => {

    it("updates the state answer onInputChange", () => {
      const preventDefaultFunction: any = jest.fn();

      expect(wrapper.state().answer).toEqual("");
 
      instance.onInputChange({
        target: {value: 123},
        preventDefault: preventDefaultFunction
      });
 
      expect(wrapper.state().answer).toEqual(123);
      expect(preventDefaultFunction).toHaveBeenCalledTimes(1);
    });

    it("updates hideAnswer onEnter", () => {
      expect(wrapper.state().hideAnswer).toEqual(true);

      instance.onEnter();

      expect(wrapper.state().hideAnswer).toEqual(false);
    });

    it("updates hideAnswer onExit", () => {
      wrapper.setState({ hideAnswer: false })

      instance.onExit();

      expect(wrapper.state().hideAnswer).toEqual(true);
    });
  });
});