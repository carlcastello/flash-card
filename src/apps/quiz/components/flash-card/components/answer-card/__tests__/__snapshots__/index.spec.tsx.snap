// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The AnswerCard the render funtion shows the answer card as FlashcardStatus.DEFAULT 1`] = `
<Styled(MuiBox)
  pt={2.5}
>
  <WithStyles(ForwardRef(Paper))
    className="paper"
    elevation={3}
  >
    <Styled(MuiBox)
      p={2.5}
    >
      <Styled(MuiBox)>
        <WithStyles(ForwardRef(Input))
          className="string"
          disableUnderline={true}
          disabled={false}
          fullWidth={true}
          margin="none"
          onChange={[Function]}
          value=""
        />
        <WithStyles(ForwardRef(Collapse))
          in={false}
          onEnter={[Function]}
          onExit={[Function]}
        >
          <Styled(MuiBox)
            pt={1.5}
          >
            <Styled(MuiBox)>
              <WithStyles(ForwardRef(Typography))
                className="typography"
                variant="subtitle2"
              >
                Answer
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Input))
                className="string"
                disableUnderline={true}
                disabled={true}
                fullWidth={true}
                margin="none"
                value=""
              />
            </Styled(MuiBox)>
          </Styled(MuiBox)>
        </WithStyles(ForwardRef(Collapse))>
      </Styled(MuiBox)>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Paper))>
</Styled(MuiBox)>
`;

exports[`The AnswerCard the render funtion shows the answer casrd as FlashcardStatus.CORRECT 1`] = `
<Styled(MuiBox)
  pt={2.5}
>
  <WithStyles(ForwardRef(Paper))
    className="paper"
    elevation={3}
  >
    <Styled(MuiBox)
      p={2.5}
    >
      <Styled(MuiBox)>
        <WithStyles(ForwardRef(Input))
          className="string"
          disableUnderline={true}
          disabled={true}
          fullWidth={true}
          margin="none"
          onChange={[Function]}
          value=""
        />
        <WithStyles(ForwardRef(Collapse))
          in={false}
          onEnter={[Function]}
          onExit={[Function]}
        >
          <Styled(MuiBox)
            pt={1.5}
          >
            <Styled(MuiBox)>
              <WithStyles(ForwardRef(Typography))
                className="typography"
                variant="subtitle2"
              >
                Answer
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Input))
                className="string"
                disableUnderline={true}
                disabled={true}
                fullWidth={true}
                margin="none"
                value=""
              />
            </Styled(MuiBox)>
          </Styled(MuiBox)>
        </WithStyles(ForwardRef(Collapse))>
      </Styled(MuiBox)>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Paper))>
</Styled(MuiBox)>
`;

exports[`The AnswerCard the render funtion shows the correct answer and removed the input when FlashcardStatus.HINT 1`] = `
<Styled(MuiBox)
  pt={2.5}
>
  <WithStyles(ForwardRef(Paper))
    className="paper"
    elevation={3}
  >
    <Styled(MuiBox)
      p={2.5}
    >
      <Styled(MuiBox)>
        <WithStyles(ForwardRef(Collapse))
          in={true}
          onEnter={[Function]}
          onExit={[Function]}
        >
          <Styled(MuiBox)
            pt={1.5}
          >
            <Styled(MuiBox)>
              <WithStyles(ForwardRef(Typography))
                className="typography"
                variant="subtitle2"
              >
                Answer
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Input))
                className="string"
                disableUnderline={true}
                disabled={true}
                fullWidth={true}
                margin="none"
                value="This is the correct answer"
              />
            </Styled(MuiBox)>
          </Styled(MuiBox)>
        </WithStyles(ForwardRef(Collapse))>
      </Styled(MuiBox)>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Paper))>
</Styled(MuiBox)>
`;

exports[`The AnswerCard the render funtion shows the correct answer when FlashcardStatus.WRONG 1`] = `
<Styled(MuiBox)
  pt={2.5}
>
  <WithStyles(ForwardRef(Paper))
    className="paper"
    elevation={3}
  >
    <Styled(MuiBox)
      p={2.5}
    >
      <Styled(MuiBox)>
        <WithStyles(ForwardRef(Input))
          className="string"
          disableUnderline={true}
          disabled={true}
          fullWidth={true}
          margin="none"
          onChange={[Function]}
          value=""
        />
        <WithStyles(ForwardRef(Collapse))
          in={true}
          onEnter={[Function]}
          onExit={[Function]}
        >
          <Styled(MuiBox)
            pt={1.5}
          >
            <Styled(MuiBox)>
              <WithStyles(ForwardRef(Typography))
                className="typography"
                variant="subtitle2"
              >
                Answer
              </WithStyles(ForwardRef(Typography))>
              <WithStyles(ForwardRef(Input))
                className="string"
                disableUnderline={true}
                disabled={true}
                fullWidth={true}
                margin="none"
                value="This is the correct answer"
              />
            </Styled(MuiBox)>
          </Styled(MuiBox)>
        </WithStyles(ForwardRef(Collapse))>
      </Styled(MuiBox)>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Paper))>
</Styled(MuiBox)>
`;
